src/lib/utils/fetchDataOnServer.ts

```diff
@@ -0,0 +1,31 @@
+import type { HttpMethod } from '@/lib/types/request';
+import { ROUTES } from '@/lib/constants/routes';
+
+export type ParsedUrl = {
+  method: HttpMethod;
+  url: string;
+  body: string;
+  headers: Record<string, string>;
+};
+
+export const fetchDataOnServer = async ({
```

Comment:
давай назовем функцию как fetchData и переиспользуем в хуке useProxyResponse

```suggestion
export const fetchData = async ({
```